### DISCUSSION TITLE
Python API Image Upload - Re-Use Images for Parts?

### DISCUSSION THREAD

**Discussion by foxforty2** [inventree/InvenTree#4550] [Q&A] [answered]
Hello,

I am exploring automatic creating of parts via the Python interface. I am able to create parts and upload images. However, when I upload the same image (Local filename: C_0805.jpg) to two parts, the file is saved on the server two times (Server filename: C_0805_xxxxxxx.jpg).

![inventree_re-use_images](https://user-images.githubusercontent.com/127667330/229285748-9d55585e-d8bb-4046-a2a3-a96180965851.png)

Is there a way to re-use already uploaded files as images for freshly created parts using the python API?

**Comment by SchrodingersGat** [comment, marked as answer]
@foxforty2 the "image sharing" functionality is not currently exposed to the API unfortunately. To achieve what you want, we would have to allow the API access to these shared images
